datafusion/optimizer/src/propagate_empty_relation.rs

```diff
@@ -149,14 +149,11 @@ fn binary_plan_children_is_empty(plan: &LogicalPlan) -> Result<(bool, bool)> {
     // all binary-plan need to deal with separately.
     match inputs.len() {
         2 => {
-            let left = inputs.get(0).unwrap();
-            let right = inputs.get(1).unwrap();
-
-            let left_empty = match left {
+            let left_empty = match inputs[0] {
```

Comment:
We could also match on `inputs` directly I think.

Reply:
Instead of on the length

Reply:
thanks @Dandandan